format-delimited/src/main/java/io/cdap/plugin/format/delimited/input/DelimitedConfig.java

```diff
@@ -188,7 +189,9 @@ public Schema getDefaultSchema(@Nullable FormatContext context) throws IOExcepti
     for (Map.Entry<String, String> entry : getFileSystemProperties().entrySet()) {
       configuration.set(entry.getKey(), entry.getValue());
     }
-    Path filePath = getFilePathForSchemaGeneration(path, regexPathFilter, configuration, job);
+//    Path filePath = getFilePathForSchemaGeneration(path, regexPathFilter, configuration, job);
+    ArrayList<Path> paths = getFilePathForSchemaGeneration(path, regexPathFilter, configuration, job);
+    Path filePath = paths.get(0);
```

Comment:
We are returning a list but only using the first result?  Why are we returning a list?

Reply:
For the Parquet use case, we do check files until we find the first one that can give us a valid schema. This is not the case for Delimited and PathTracking implementations.